docs/markdown/30-testing/01-others.md

```diff
@@ -13,4 +13,88 @@ If both cases, you can use **bats** to help you write a test suite for your acti
 
 # Testing Shell script with Bats
 
-TODO example
+## Test folder
+
+```yaml
+.github/
+    actions/
+        runs-using-docker/
+            cleanup.sh
+            main.sh
+            ...
+test/
+    bats/               <- submodule
+    test_helper/
+        bats-support/   <- submodule
+        bats-assert/    <- submodule
+    test.bats
+    ...
+
+```
+
+
+##--##
+
+# Testing Shell script with Bats
+
+## **test.bats** 
+
+```
+setup() {
+    load 'test_helper/bats-support/load'
+    load 'test_helper/bats-assert/load'
+
+    DIR="$( cd "$( dirname "$BATS_TEST_FILENAME" )" >/dev/null 2>&1 && pwd )"
+    export KEY=tata
+    PATH="$DIR/../.github/actions/runs-using-docker:$PATH"
+}
+
+@test "check main.sh is working" {
+    run main.sh toto
+    assert_output 'Doing stuff... toto (tata)'
+}
+
+@test "check cleanup.sh is working" {
+    run cleanup.sh toto
+    assert_output 'Cleaning up... toto (tata)'
+}
+
+@test "check setup.sh is working" {
+    run setup.sh toto
+    assert_output 'Setting up... toto (tata)'
+}
+```
+
+- Easy way to writte action fgollowing the TDD method
+- Required github submodule to work : 
+    - `test/bats` 👉 [**Go to reference** 🔗](https://github.com/bats-core/bats-core.git)
+    - `test/test_helper/bats-support` 👉 [**Go to reference** 🔗](https://github.com/bats-core/bats-support.git)
+    - `test/test_helper/bats-assert` 👉 [**Go to reference** 🔗](https://github.com/bats-core/bats-assert.git)
```

Comment:
Move this part to the previous slide so that the flow will be 

- bats exist for shell testing
- setup bats into your action
- write a bats file
- run bats inside your GHA workflow

Reply:
Done